### ISSUE TITLE
Uncaught Error: EEXIST, file already exists

### ISSUE THREAD

**Issue by jmm** [atom/atom#5640]
[Enter steps to reproduce below:]

1. Try to save a modified file.

Out of the blue trying to save a file triggered this error, after opening, editing, saving files from the same place for a while. The file is on a samba share. This seems similar to #5518, but in my case there's no sign of the disk having become unavailable. I can open and save other files from the same folder on the same share. But even after that I still can't save the original file.

**Atom Version**: 0.179.0
**System**: Microsoft Windows 7 Home Premium 
**Thrown From**: Atom Core

### Stack Trace

Uncaught Error: EEXIST, file already exists '\\server.\share\dir\dir\src\rsrc\model'

```
At C:\Program Files (x86)\GitHub Atom\resources\app\src\pane.js:782

Error: EEXIST, file already exists '\\server.\share\dir\dir\src\rsrc\model'
  at Error (native)
  at Object.fs.mkdirSync (fs.js:752:18)
  at Function.sync (C:\Program Files (x86)\GitHub Atom\resources\app\node_modules\mkdirp\index.js:55:12)
  at Object.fsPlus.writeFileSync (C:\Program Files (x86)\GitHub Atom\resources\app\node_modules\fs-plus\lib\fs-plus.js:242:14)
  at File.module.exports.File.writeFileSync (C:\Program Files (x86)\GitHub Atom\resources\app\node_modules\pathwatcher\lib\file.js:234:19)
  at File.module.exports.File.writeFileWithPrivilegeEscalationSync (C:\Program Files (x86)\GitHub Atom\resources\app\node_modules\pathwatcher\lib\file.js:304:21)
  at File.module.exports.File.write (C:\Program Files (x86)\GitHub Atom\resources\app\node_modules\pathwatcher\lib\file.js:293:12)
  at TextBuffer.module.exports.TextBuffer.saveAs (C:\Program Files (x86)\GitHub Atom\resources\app\node_modules\text-buffer\lib\text-buffer.js:970:17)
  at TextBuffer.module.exports.TextBuffer.save (C:\Program Files (x86)\GitHub Atom\resources\app\node_modules\text-buffer\lib\text-buffer.js:958:19)
  at TextEditor.module.exports.TextEditor.save (C:\Program Files (x86)\GitHub Atom\resources\app\src\text-editor.js:620:26)
  at Pane.module.exports.Pane.saveItem (C:\Program Files (x86)\GitHub Atom\resources\app\src\pane.js:542:18)
  at Pane.module.exports.Pane.saveActiveItem (C:\Program Files (x86)\GitHub Atom\resources\app\src\pane.js:525:19)
  at Workspace.module.exports.Workspace.saveActivePaneItem (C:\Program Files (x86)\GitHub Atom\resources\app\src\workspace.js:679:35)
  at atom-workspace.atom.commands.add.core:save (C:\Program Files (x86)\GitHub Atom\resources\app\src\workspace-element.js:295:30)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (C:\Program Files (x86)\GitHub Atom\resources\app\src\command-registry.js:243:29)
  at C:\Program Files (x86)\GitHub Atom\resources\app\src\command-registry.js:3:61
  at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (C:\Program Files (x86)\GitHub Atom\resources\app\node_modules\atom-keymap\lib\keymap-manager.js:558:16)
  at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (C:\Program Files (x86)\GitHub Atom\resources\app\node_modules\atom-keymap\lib\keymap-manager.js:396:22)
  at HTMLDocument.module.exports.WindowEventHandler.onKeydown (C:\Program Files (x86)\GitHub Atom\resources\app\src\window-event-handler.js:172:20)

```

### Commands

```
 16x -8:02.8 core:move-down (atom-text-editor.editor.is-focused)
 30x -8:01.5 core:move-up (atom-text-editor.editor.is-focused)
 25x -7:57.4 core:move-down (atom-text-editor.editor.is-focused)
 26x -7:53.8 core:move-up (atom-text-editor.editor.is-focused)
 15x -7:35.7 core:move-down (atom-text-editor.editor.is-focused)
     -7:31.6 editor:newline (atom-text-editor.editor.is-focused)
     -7:31.0 core:save (atom-text-editor.editor.is-focused)
 20x -7:29.1 core:move-down (atom-text-editor.editor.is-focused)
     -7:16.8 core:save (atom-text-editor.editor.is-focused)
  3x -7:07.9 core:move-up (atom-text-editor.editor.is-focused)
  2x -7:07.3 core:save (atom-text-editor.editor.is-focused)
     -0:15.3 window:toggle-dev-tools (atom-text-editor.editor.is-focused)
     -0:05.2 find-and-replace:show (atom-text-editor.editor.is-focused)
     -0:04.5 editor:consolidate-selections (atom-text-editor.editor.mini)
     -0:04.5 core:cancel (atom-text-editor.editor.mini)
     -0:03.8 core:save (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "ignoredNames": [
      ".git",
      ".hg",
      ".svn",
      ".DS_Store",
      "Thumbs.db",
      "bower_components",
      "node_modules"
    ]
  },
  "editor": {
    "fontSize": 13,
    "autoIndentOnPaste": false,
    "softWrap": true,
    "invisibles": {},
    "softWrapAtPreferredLineLength": true
  }
}
```

### Installed Packages

```coffee
# User
No installed packages

# Dev
No dev packages
```

**Comment by jmm** [comment]
I realized it's because the file actually doesn't exist, because a git branch that doesn't contain the file is checked out. But I switch branches all the time and usually Atom automatically closes files that don't exist in the checked out branch. Maybe I accidentally edited the file (without saving) while its branch was checked out and then checked out the other branch. In a case like that, is this the expected behavior, that it fails with this error instead of just saving the file at the established path?

**Comment by jmm** [comment]
Atom has also gotten really flaky in general about updating files that have changed on disk, e.g. when switching git branches. I would prefer that there be an option to prompt before updating a file that has changed on disk, but now it often just silently doesn't do anything (update or prompt) when a file has changed on disk. And there's *still* not even a `File > Revert` command. I think I started noticing this problem when I updated from whatever to 0.174.0.
